scripts/generateMetaCallsForExtension.ts

```diff
@@ -0,0 +1,34 @@
+import { Api } from '../packages/api/src/Api';
+import Types, { typesBundle } from '../packages/types/src/interfaces/injects';
+import { Metadata } from '@polkadot/metadata';
+import * as fs from 'fs';
+import process from "process";
+
+async function getMetaCalls(apiList: Api[]) {
+  let metaMapObject = apiList.reduce(function (accumulator, currentApi) {
+    let specVersion = currentApi.runtimeVersion.specVersion.toString();
+    let genesisHash = currentApi.genesisHash.toString();
+    const metaFetched = new Metadata(currentApi.registry, currentApi.runtimeMetadata.toHex());
+    const metaCalls = Buffer.from(metaFetched.asCallsOnly.toU8a()).toString('base64');
+    accumulator[`${genesisHash}-${specVersion}`] = metaCalls;
+    return accumulator;
+  }, {});
+  const metaDataJSON = {"metaCalls": metaMapObject};
+  console.log(`Writing static metadata to extension-releases/metadata.json`);
+  fs.writeFileSync("extension-releases/metadata.json", JSON.stringify(metaDataJSON) , 'utf8');
+  process.exit()
+}
+
+// CENNZnet extension will listen to the metadata generated by this script to update its metadata
+async function generateMetaCallsForExtensionReleases() {
+  const providerAzalea = 'wss://cennznet.unfrastructure.io/public/ws';
+  const apiAzalea = await Api.create({provider: providerAzalea, types: Types});
+  const providerNikau = 'wss://nikau.centrality.me/public/ws';
+  const apiNikau = await Api.create({provider: providerNikau, types: Types});
+  const providerRata = 'wss://kong2.centrality.me/public/rata/ws';
+  const apiRata = await Api.create({provider: providerRata, types: Types});
+  const apiLocal = await Api.create({types: Types});
+  await getMetaCalls([apiAzalea, apiNikau, apiRata, apiLocal]);
```

Comment:
it should be possible to publish this for a runtime version before it goes live 💭

Reply:
ya... will get the spec version and metaCalls for the local chain (with latest docker image) and then use it for all the chains... genesis hash would be queried for individual chain...